cmd/serve/option.go

```diff
@@ -0,0 +1,127 @@
+// Copyright 2020 Authors of Hubble
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package serve
+
+import (
+	"net"
+	"os"
+	"strings"
+
+	"github.com/cilium/hubble/pkg/api"
+	v1 "github.com/cilium/hubble/pkg/api/v1"
+	"github.com/cilium/hubble/pkg/server"
+	"golang.org/x/sys/unix"
+	"google.golang.org/grpc/health"
+	healthpb "google.golang.org/grpc/health/grpc_health_v1"
+)
+
+// DefaultOptions is the reference point for default values.
+var DefaultOptions = Options{
+	Listeners: make(map[string]net.Listener),
+}
+
+// Options stores all the configuration values for the hubble server.
+type Options struct {
+	Listeners       map[string]net.Listener
+	HealthService   *health.Server
+	ObserverService *server.GRPCServer
+}
+
+// Option customizes then configuration of the hubble server.
+type Option func(o *Options) error
+
+// WithListeners configures listeners. Addresses that are prefixed with
+// 'unix://' are assumed to be UNIX domain sockets, in which case appropriate
+// permissions are tentatively set and the group owner is set to socketGroup.
+// Otherwise, the address is assumed to be TCP.
+func WithListeners(addresses []string, socketGroup string) Option {
+	return func(o *Options) error {
+		var err error
+		defer func() {
```

Comment:
can you inline this as an if towards the end of the function rather than doing a defer?

Reply:
Done.